UnicornEngine/video/include/unicorn/video/vulkan/VkMesh.hpp

```diff
@@ -91,30 +91,45 @@ class VkMesh
     */
     uint32_t IndicesSize() const;
 
+    /**
+     * @brief Checks if needed to render colored mesh
+     * @return true if mesh colored and false if not
+     */
     bool IsColored() const;
 
+    /**
+     * @brief Checks if wireframe mode enabled for this mesh
+     * @return true if mesh needs to render in wireframe mode, false if not
+     */
     bool IsWired() const;
 
+    /**
+     * @brief Returns color of mesh
+     * @return color of mesh
+     */
     glm::vec3 GetColor() const;
 
-    /*
-     * @brief Material in vulkan is combination of descriptor set and binded data
-     * //TODO
+    /**
+     * @brief Updates data if material of mesh was updated
      */
-    void SetMaterialHandle(uint32_t handle);
+    void OnMaterialUpdated();
 
-    void OnMaterialUpdated(Mesh* mesh);
+    /**
+     * @brief Material in vulkan is combination of descriptor set and binded data   
+     */
+    uint32_t materialHandle;
 
-    uint32_t GetMaterialHandle() const;
     /**
      * @brief Signal for command buffer reallocation
      */
     wink::signal<wink::slot<void(VkMesh*)>> ReallocatedOnGpu;
+
+    /**
+    * @brief Signal for uniform update and commnd buffer reallocation
```

Comment:
`command`
Also what is `uniform`? The signal is called MaterialUpdated, but you say it is triggered on some uniform update and command buffer realloc?

Reply:
I wrote "for uniform update", not "on uniform". Uniform is sampler, which binded to shader (descriptor set), which we need to set again and generate command buffers.

Reply:
I assume mesh doesn't know about uniforms or command buffers. These external details shall be avoided when describing class contents.

Reply:
Agreed